src/main/java/servlets/OAuthCallbackServlet.java

```diff
@@ -61,6 +64,13 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       response.setStatus(400);
       return;
     }
+        
+    String sessionOauthState = (String) request.getSession().getAttribute(OAuthConstants.SHEETS_SESSION_KEY);
+
+    if (!state.equals(sessionOauthState)) {
+      response.setStatus(401);
+      return;
+    }
```

Comment:
can you be explicit what the 401 status means?

Reply:
+1

Reply:
Sure, I'll add a comment about it. A 401 error is a unauthorized client error, which means the request was lacking valid auth credentials.